File: dart/7kyu/alternate_capitalization.dart
// https://www.codewars.com/kata/59cfc000aeb2844d16000075

List<String> capitalize(String x) {
  StringBuffer lowerString = StringBuffer();
  StringBuffer upperString = StringBuffer();
  String lower;
  String upper;
  for (int i = 0; i < x.length; i++) {
    lower = x[i].toLowerCase();
    upper = x[i].toUpperCase();
    if (i % 2 == 0) {
      lowerString.write(lower);
      upperString.write(upper);
    } else {
      lowerString.write(upper);
      upperString.write(lower);
    }
  }
  return [upperString.toString(), lowerString.toString()];
}

void main() {
  print(capitalize("abcdef"));
}

File: clojure/7kyu/alternate_capitalization.clj
; https://www.codewars.com/kata/59cfc000aeb2844d16000075
(println "Alternate Capitaliztion")



(defn solve [s]
 (reduce (fn [x i] (println i) ) [] (map (fn [i] [(Character/toUpperCase i) (Character/toLowerCase i)]) s)))

(solve "abcdef")
(partition 0 2 "12345")
(map (fn [i] [(Character/toUpperCase i) (Character/toLowerCase i)]) "abcdef")
(reduce str "aaa")
(def t (cycle [[0 1] [1 0]]))
(take 10 t)

(for [i [[\A \a] [\B \b]]
      x t]
  (println x))

(= (solve "abcdef") ["AbCdEf" "aBcDeF"])
(= (solve "codewars" ["CoDeWaRs" "cOdEwArS"]))
